src/timezone_tools/_converter.py

```diff
@@ -269,7 +273,7 @@ def end_of_month(self, datetime: datetime_.datetime) -> datetime_.datetime:
         )
 
     def first_day_of_month(
-        self, datetime: datetime_.datetime
+        self, datetime: datetime_.datetime | datetime_.date
     ) -> datetime_.date:
         """Find the date of the first day of this month in this timezone.
```

Comment:
Thinking about this, there's a risk here. If `datetime` is a naive datetime, we'd raise an exception because we don't know what moment in time it refers to, so we don't know which month that was in in our timezone. The same is sort of true for dates: they are inherently naive. A date refers to a 49 hour-long period, which could quite reasonably be a moment in time in one of two months.

If a caller wants to get the first _date_ of a month, the can just call `.replace(day=1)` on it. If they're trying to be precise about timezones, I don't think this is the right API for them. They'd be better off getting that date then passing it to `midnight`.

Reply:
Hm... I see your point. This would require the users to really pay attention to the `date` they're passing and it's probably safer to be strict and require a localized datetime. I was kinda hoping to avoid calls like `paris_time.last_day_of_month(paris_time.midnight(date))` but in the end it only safes a few chars for a lot more uncertainty.